tests/ops/op_math/test_sum.py

```diff
@@ -107,109 +114,154 @@ def compare_and_expand_mat(mat1, mat2):
     return smaller_mat, larger_mat
 
 
+@pytest.mark.parametrize("sum_method", [sum_using_dunder_method, op_sum])
 class TestInitialization:
+    """Test the initialization."""
+
     @pytest.mark.parametrize("id", ("foo", "bar"))
-    def test_init_sum_op(self, id):
-        sum_op = op_sum(qml.PauliX(wires=0), qml.RZ(0.23, wires="a"), do_queue=True, id=id)
+    def test_init_sum_op(self, id, sum_method):
+        """Test the initialization of a Sum operator."""
+        sum_op = sum_method(qml.PauliX(wires=0), qml.RZ(0.23, wires="a"), do_queue=True, id=id)
 
         assert sum_op.wires == Wires((0, "a"))
         assert sum_op.num_wires == 2
         assert sum_op.name == "Sum"
-        assert sum_op.id == id
+        if sum_method.__name__ == op_sum.__name__:
+            assert sum_op.id in id
```

Comment:
This test passes because `"my_id" in "my_id" == True` since a string is always a substring of itself haha 😄 

```suggestion
            assert sum_op.id == id
```

Reply:
Fixed in new PR https://github.com/PennyLaneAI/pennylane/pull/2825 (see comment https://github.com/PennyLaneAI/pennylane/pull/2807#issuecomment-1185445582)